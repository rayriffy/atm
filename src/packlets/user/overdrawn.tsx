import { HStack, Text } from '@chakra-ui/react'
import { Icon } from 'react-iconify-icon-wrapper'

import { useBalanceAtom } from '$context/balanceAtom'

export const Overdrawn = () => {
  const balance = useBalanceAtom()

  if (balance > 0) return null

  return (
    <HStack bg="orange.600" color="white" rounded="md" px={2.5} py={2}>
      <Icon icon="lucide:alert-triangle" width={16} height={16} />
      <Text fontWeight="semibold" fontSize="sm" lineHeight={1}>
        Account overdrawn
      </Text>
    </HStack>
  )
}
