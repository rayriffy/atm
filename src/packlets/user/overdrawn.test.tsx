import { render } from '@testing-library/react'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import { Overdrawn } from './overdrawn'

const { useBalanceAtomSpy } = vi.hoisted(() => ({
  useBalanceAtomSpy: vi.fn().mockReturnValue(0),
}))

vi.mock('$context/balanceAtom', () => ({
  useBalanceAtom: useBalanceAtomSpy,
}))

describe('decodeError', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('should render nothing if balance healty', () => {
    useBalanceAtomSpy.mockReturnValue(0)

    const { container } = render(<Overdrawn />)

    expect(container.firstChild).toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('should render overdrawn badge if balance goes into minus', () => {
    useBalanceAtomSpy.mockReturnValue(-1)

    const { container, getByText } = render(<Overdrawn />)

    expect(getByText('Account overdrawn')).toBeDefined()

    expect(container).toMatchSnapshot()
  })
})
