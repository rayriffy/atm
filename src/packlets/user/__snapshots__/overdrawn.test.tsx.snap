// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`decodeError > should render nothing if balance healty 1`] = `<div />`;

exports[`decodeError > should render overdrawn badge if balance goes into minus 1`] = `
<div>
  <div
    class="chakra-stack css-1so6s9z"
  >
    <iconify-icon
      height="16"
      icon="lucide:alert-triangle"
      width="16"
    />
    <p
      class="chakra-text css-nrviwf"
    >
      Account overdrawn
    </p>
  </div>
</div>
`;
